src/tools/prte/prte.c

```diff
@@ -850,6 +852,11 @@ int main(int argc, char *argv[])
         free(tval);
     }
 
+    /* Was a rankfile specified? */
```

Comment:
I'm a little puzzled here. Schizo is treating the `--rankfile` option as deprecated and pushing it into the `--map-by` option as a qualifier that the `rmaps` framework subsequently parses. So why is `prte` looking at the `rankfile` option here?

I imagine we will eventually have to do something about that `prte_rankfile` global as we don't want that to apply to every job, and we can't let any subsequent `prun` invocations overwrite it due to race conditions. For now, though, I should think this would fix the immediate problem.

Reply:
Do we want to deprecate `--rankfile`? Or maybe make it an OMPI-only option (since they wanted to keep it [last we discussed](https://github.com/open-mpi/ompi/wiki/Webex-affinity-discussions-2019-09#conclusions) it with them)?

For `prte_rankfile` that's a second effort aligning with Issue #720 . So that'll happen after this.

Reply:
If they want to retain it, then we should push it into the OMPI component. We probably want to retain a "rankfile" qualifier in `map-by` for the PRRTE users, so I think all we want to do is move the `--rankfile` code to the `ompi` component and not treat it as deprecated - just silently translate to the `--map-by` qualifier.

My question about `prte_rankfile` really had two elements. No issue with pushing off making it non-global, but I still don't understand why `prte` is looking for `--rankfile` directly and setting `prte_rankfile`. Given that `schizo` just converted it to a `--map-by` qualifier that the `rmaps` framework will handle, I should think `prte` wouldn't itself look for that option at all - yes?

Reply:
I put that check in prte.c originally before I had the code to handle --map-by rankfile working correctly. I just tried deleting the prte.c code and found that it's not required, so I can remove it.

Is there a way to silently translate --rankfile rankfile1 to --map-by rankfile:rankfile1?

Also, is --rankfile to be recognized only with the ompi personality, meaning it's invalid with prterun with no --personality flag, and that the only way a rankfile will be recognized it prterun and no --personality flag is as --map-by rankfile:rankfile1?

Reply:
> so I can remove it.

Thanks!

> --rankfile to be recognized only with the ompi personality, meaning it's invalid with prterun with no --personality flag, and that the only way a rankfile will be recognized it prterun and no --personality flag is as --map-by rankfile:rankfile1

I think that's the idea, yes. We really do want to reduce the number of cmd line options as otherwise `prterun --help` becomes a tome. We'd like to instead have only a few options and refer people to the upcoming docs pages.